artifactory/system.go

```diff
@@ -33,6 +40,1112 @@ func (v Versions) String() string {
+	GcConfig                  *GcConfig                  `yaml:"gcConfig,omitempty" xml:"gcConfig,omitempty"`
+	CleanupConfig             *CleanupConfig             `yaml:"cleanupConfig,omitempty" xml:"cleanupConfig,omitempty"`
+	VirtualCacheCleanupConfig *VirtualCacheCleanupConfig `yaml:"virtualCacheCleanupConfig,omitempty" xml:"virtualCacheCleanupConfig,omitempty"`
+	QuotaConfig               *QuotaConfig               `yaml:"quotaConfig,omitempty" xml:"quotaConfig,omitempty"`
+	SystemMessageConfig       *SystemMessageConfig       `yaml:"systemMessageConfig,omitempty" xml:"systemMessageConfig,omitempty"`
+	FolderDownloadConfig      *FolderDownloadConfig      `yaml:"folderDownloadConfig,omitempty" xml:"folderDownloadConfig,omitempty"`
+	TrashcanConfig            *TrashcanConfig            `yaml:"trashcanConfig,omitempty" xml:"trashcanConfig,omitempty"`
+	ReplicationsConfig        *ReplicationsConfig        `yaml:"replicationsConfig,omitempty" xml:"replicationsConfig,omitempty"`
+	SumoLogicConfig           *SumoLogicConfig           `yaml:"sumoLogicConfig,omitempty" xml:"sumoLogicConfig,omitempty"`
+	ReleaseBundlesConfig      *ReleaseBundlesConfig      `yaml:"releaseBundlesConfig,omitempty" xml:"releaseBundlesConfig,omitempty"`
+	SignedUrlConfig           *SignedUrlConfig           `yaml:"signedUrlConfig,omitempty" xml:"signedUrlConfig,omitempty"`
+	DownloadRedirectConfig    *DownloadRedirectConfig    `yaml:"downloadRedirectConfig,omitempty" xml:"downloadRedirectConfig,omitempty"`
+	Security                  *Security                  `yaml:"security,omitempty" xml:"security,omitempty"`
+	Backups                   *Backups                   `yaml:"backups,omitempty" xml:"backups>backup,omitempty"`
+	Proxies                   *Proxies                   `yaml:"proxies,omitempty" xml:"proxies>proxy,omitempty"`
+	ReverseProxies            *ReverseProxies            `yaml:"reverseProxies,omitempty" xml:"reverseProxies>reverseProxy,omitempty"`
+	PropertySets              *PropertySets              `yaml:"propertySets,omitempty" xml:"propertySets>propertySet,omitempty"`
+	RepoLayouts               *RepoLayouts               `yaml:"repoLayouts,omitempty" xml:"repoLayouts>repoLayout,omitempty"`
+	BintrayApplications       *BintrayApplications       `yaml:"bintrayApplications,omitempty" xml:"bintrayApplications>bintrayApplication,omitempty"`
+	LocalRepositories         *[]LocalRepository         `yaml:"-" xml:"localRepositories>localRepository,omitempty"`
+	RemoteRepositories        *[]RemoteRepository        `yaml:"-" xml:"remoteRepositories>remoteRepository,omitempty"`
+	VirtualRepositories       *[]VirtualRepository       `yaml:"-" xml:"virtualRepositories>virtualRepository,omitempty"`
+	LocalReplications         *[]Replication             `yaml:"-" xml:"localReplications>localReplication,omitempty"`
+	RemoteReplications        *[]Replication             `yaml:"-" xml:"remoteReplications>remoteReplication,omitempty"`
```

Comment:
I noticed we're not allowing these fields to be serialized in YAML due to the `-` tag.

This peaked my curiosity so was this intentional?

Reply:
Oops, sorry, I had commented this in a previous commit but stripped it out. 

I didn't think it was a good idea to provide the ability to CRUD repos and replications via the YAML since this functionality is provided with the existing API methods and this is generally a safer option as doing it via the YAML can render Artifactory in an unusable state. For example, it's possible to overwrite a repository's package type (warnings all over the [doc](https://www.jfrog.com/confluence/display/JFROG/Artifactory+YAML+Configuration)).